rust/cas_client/src/data_transport.rs

```diff
@@ -335,9 +328,13 @@ impl DataTransport {
             .retry_strategy
             .retry(
                 || async {
-                    let req = self
+                    let full_size = data.len();
+                    // compression of data to be done here, for now none.
+                    let mut req = self
                         .setup_request(Method::POST, prefix, hash, Some(data.to_owned()))
                         .map_err(RetryError::from)?;
+                    let headers = req.headers_mut();
+                    headers.insert(CAS_INFLATED_SIZE_HEADER, HeaderValue::from(full_size));
```

Comment:
Is this for verification of content on the cas server?

Reply:
We need the total inflated size for decompression. We can either supply it ourselves like this, or alternatively we can  pack it into the compressed content, but according to YL this could block us from doing decompression while streaming. So we'll use this instead.